www/src/templates/template-api-markdown.js

```diff
@@ -0,0 +1,185 @@
+import { MDXRenderer } from "gatsby-plugin-mdx"
+import { mediaQueries } from "../gatsby-plugin-theme-ui"
+
+// API Rendering Stuff
+import { sortBy } from "lodash-es"
+
+import Layout from "../components/layout"
+import { itemListDocs } from "../utils/sidebar/item-list"
+import MarkdownPageFooter from "../components/markdown-page-footer"
+import DocSearchContent from "../components/docsearch-content"
+import FooterLinks from "../components/shared/footer-links"
+import Breadcrumb from "../components/docs-breadcrumb"
+import Container from "../components/container"
+import PrevAndNext from "../components/prev-and-next"
+import APIReference, { APIContents } from "../components/api-reference"
+
+const normalizeGatsbyApiCall = array =>
+  array.map(entry => {
+    const codeLocation =
+      entry.nodes.length > 1
+        ? entry.nodes.map(l => {
+            return {
+              file: l.file,
+              start: { line: l.codeLocation.start.line },
+              end: { line: l.codeLocation.end.line },
+            }
+          })
+        : {
+            file: entry.nodes[0].file,
+            start: { line: entry.nodes[0].codeLocation.start.line },
+            end: { line: entry.nodes[0].codeLocation.end.line },
+          }
+
+    return { name: entry.name, codeLocation }
+  })
+
+const mergeFunctions = (data, context) => {
+  const normalized = normalizeGatsbyApiCall(data.nodeAPIs.group)
+
+  const docs = data.jsdoc.nodes.reduce((acc, node) => {
+    const doc = node.childrenDocumentationJs
+      .filter(def => def.kind !== `typedef` && def.memberof)
+      .map(def => {
+        if (!context.apiCalls) {
+          // When an api call list is not available, the line numbers from jsdoc
+          // might be useful. Just for actions.mdx right now.
+          def.codeLocation.file = node.relativePath
+          if (!def.codeLocation.file) {
+            def.codeLocation = null
+          }
+        } else {
+          // API pages having apiCalls did not query for this in the page query,
+          // so just remove it instead. Having one that returns nothing supresses
+          // documentation git links completely.
+          def.codeLocation = null
+        }
+        return def
+      })
+    return acc.concat(doc)
+  }, [])
+
+  let funcs = sortBy(docs, func => func.name)
+
+  const mergedFuncs = funcs.map(func => {
+    return {
+      ...func,
+      ...normalized.find(n => n.name === func.name),
+    }
+  })
+
+  return mergedFuncs
+}
+
+export default function APITemplate({ data, location, pageContext }) {
+  const { next, prev } = pageContext
+  const page = data.mdx
+
+  // Cleanup graphql data for usage with API rendering components
+  const mergedFuncs = mergeFunctions(data, pageContext)
+
+  return (
+    <React.Fragment>
+      <Helmet>
+        <title>{page.frontmatter.title}</title>
+        <meta name="description" content={page.excerpt} />
+        <meta property="og:description" content={page.excerpt} />
```

Comment:
The excerpt here is auto generated, but would be good to be able to provide custom one.

As example - currently for "SSR APIs" we have:

> Documentation on APIs related to server side rendering during Gatsby's build process

With this PR it changes to:

> Usage Implement any of these APIs by exporting them from a file named  gatsby-ssr.js  in the root of your project.

So maybe optional frontmatter field to define the description? (as we had it hardcoded previously)